sources/cfsignal/src/system_check.rs

```diff
@@ -0,0 +1,60 @@
+use crate::error::{self, Result};
+use log::trace;
+use snafu::ResultExt;
+use std::process::Command;
+
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub(crate) struct SystemHealth {
+    /// Whether or not the system is healthy.
+    pub(crate) is_healthy: bool,
+    /// The system status description.
+    pub(crate) status: String,
+    /// The system status code.
+    pub(crate) exit_code: i32,
+}
+
+pub(crate) struct SystemCheck {}
+
+impl SystemCheck {
+    pub fn system_running(&self) -> Result<SystemHealth> {
+        let mut result = is_system_running()?;
+        // remove trailing newline
+        result.stdout.pop();
+        Ok(SystemHealth {
+            is_healthy: result.is_success(),
+            status: result.stdout,
+            exit_code: result.exit,
+        })
+    }
+}
+
+struct Outcome {
```

Comment:
These abstractions are perhaps not as useful here as they are in metricdog (which inspired them). They are harmless though so I don't see a need to eliminate these structs and functions.